lib/Module/IntrinsicCleaner.cpp

```diff
@@ -49,6 +49,54 @@ bool IntrinsicCleanerPass::runOnModule(Module &M) {
   return dirty;
 }
 
+llvm::Constant* IntrinsicCleanerPass::createConstantVector(llvm::LLVMContext& ctx, 
+  llvm::Value& constantVector, opMode mode) { 
+  llvm::Constant* pConstant = nullptr;
+  llvm::ConstantInt* pCInt = nullptr;
+  Type* pType = nullptr;
+  Type* elementType = nullptr;
+  unsigned elementNum = 0;
+  unsigned elementBw = 0;
+  SmallVector<llvm::Constant*, 8> CElts;
+
+  pType = constantVector.getType();
+  if (!pType) { 
+    return nullptr;
+  }
+  elementType = pType->getVectorElementType();
+  if (!elementType) { 
+    return nullptr;
+  }
+  elementNum = pType->getVectorNumElements();
+  elementBw = elementType->getPrimitiveSizeInBits();
+  if (!elementNum || !elementBw) { 
+    return nullptr;
+  } 
+  if (mode == opMode::zeroValue) { 
```

Comment:
Use `switch` and `case` to handle enum.
For the default case, use `assert` again for defensive programming.